core/io/xml_parser.cpp

```diff
@@ -32,12 +32,18 @@
 
 #include "core/string/print_string.h"
 
+#include <ctype.h>
+
 //#define DEBUG_XML
 
 static inline bool _is_white_space(char c) {
 	return (c == ' ' || c == '\t' || c == '\n' || c == '\r');
 }
 
+static inline bool _is_punctuation(char c) {
+	return ((c == ',') || (c == "’") || (c == '\'') || (c == '"') || (c == '?') || (c == '!') || (c == "-") || (c == "—") || (c == '(') || (c == '[') || (c == '{') || (c == '.') || (c == ':') || (c == ';'));
```

Comment:
You are comparing against a string here, eg "-", instead of '-'.

Reply:
Thank you for your feedback, I am currently working on it